core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -376,7 +376,7 @@ private static RexNode canonizeNode(RexBuilder rexBuilder, RexNode condition) {
         return rexBuilder.makeCall(call.getOperator(), left, right);
       }
 
-      RexNode newCall = rexBuilder.makeCall(call.getOperator(), right, left);
+      RexNode newCall = rexBuilder.makeCall(call.getOperator(), left, right);
```

Comment:
I believe it is not an error.
The reordering is intentional: note the comparison in the previous line

Reply:
Thanks for review ,I have a testcase that can prove it and I will open an another PR .

Reply:
Please create a branch for the PR next time. PRs from master branches do not really work well.